polyfill/lib/calendar.mjs

```diff
@@ -633,18 +633,20 @@ const nonIsoHelperBase = {
   adjustCalendarDate(calendarDate, cache, overflow /*, fromLegacyDate = false */) {
     if (this.calendarType === 'lunisolar') throw new RangeError('Override required for lunisolar calendars');
     this.validateCalendarDate(calendarDate);
-    const largestMonth = this.monthsInYear(calendarDate, cache);
-    let { month, year, eraYear, monthCode } = calendarDate;
```

Comment:
Are you sure this isn't intentionally controlling the order of calling any getters on `calendarDate`?

Reply:
> with a shape that varies based on the method

More specifically, all "calendarDate" objects have the shape above, but depending on the method and where the input comes from, some properties may be missing vs. present. For example, if the input to the method commented above is coming from user calls, then it will match the input of `Calendar.dateFromFields`.  If the input is internally generated, it might only be YMD fields. Other methods may only care about some subset of the fields. For example, the `inLeapYear` methods of these `*Helper` objects only care about the `year` property and ignores the rest. 

The TS polyfill (especially in https://github.com/js-temporal/temporal-polyfill/pull/109 makes this all a lot clearer because everything is now typed.